Added rudimentary debian packaging information.

uroskn opened the pull request:
This should allow anyone to build a basic installable .deb package out of checkout of this repository.

I introduced basic Debian packaging files to the repository, providing an initial setup that allowed me to build a simple installable .deb package directly from a checkout - meant for environments where .deb packages are preferred (e.g. Debian, Ubuntu).

Tests during packaging are skipped, as I've noticed `--all-plugins` test fails completely on my machine.

scottchiefbaker commented:
This looks good to me because it's almost entirely isolated to the `debian/` directory and shouldn't affect much else. 

Can you elaborate on the "`--all-plugins` test fails" on my machine? Dool doesn't have any unit tests that I'm aware of?

scottchiefbaker commented:
Can you double-check the `next` branch as I landed all the Debian stuff there. I moved the Debian specific stuff into the `packaging/` directory with the other package specific stuff, and updated the `Makefile`. As I don't have a way to test this, can you test and let me know if I did it correctly?

uroskn commented:
Sorry for the tad delayed response.

>Can you elaborate on the "--all-plugins test fails" on my machine? 

Sure, Debian packaging will try to run `make test` if it finds this rule in a Makefile of a project. This rule is indeed defined in the dool Makefile and essentially runs `./dool -t --all-plugins 1 5`.

The `battery-remain` plugin in dool fails on my machine, causing `make` to exit with a non-zero code, which Debian packaging interprets as a failed test, halting the packaging process.

>Can you double-check the next branch as I landed all the Debian stuff there. I moved the Debian specific stuff into the packaging/ directory with the other package-specific materials, and updated the Makefile. As I don't have a way to test this, can you test and let me know if I did it correctly?

I've reviewed it, and these changes do break the build.

To my knowledge, `dpkg-buildpackage` requires the `debian` directory to be in the source root of the project. 

I’m not aware of any simple mechanism that would allow `dpkg-buildpackage` to locate Debian-specific files elsewhere with current structure (to my knowledge, but I'm not familiar with every switch and setting Debian build process has). 

`cd`-ing to different directory beforehand (as it was done in Makefile) will then simply cause `dpkg-buildpackage` to not find correct source tree, failing in a different way.

scottchiefbaker commented:
Sorry for the delay in getting back to you on this. I'd like to keep the root of this distro "clean" and keep the package stuff in it's own directory. You build the majority of the debian stuff, what if instead of requiring it to be in the root directory there is a `build.sh` in the packaging folder which copies all the appropriate files to the needed places in a temp directory and builds the `.deb` from there? 

I'd do it myself but I have zero experience building things on Debian.
